### ISSUE TITLE
[Components] Add A11Y tests to check labels, values, hints and traits of components

### ISSUE THREAD

**Issue by pylapp** [Orange-OpenSource/ouds-ios#1079]
# Context

In OUDS we define accessibility information for the components to as to vocalize them in the best way with _Voice Over_.

Do do that, we can defined several properties:
- accessibility label
- accessibility hint
- accessibility value
- accessibility traits

However, even if we can do human tests with _Voice Over_, automate them will improve regressions tests.

# Issue

Nevertheless we face troubles: it seems *Voice Over* vocalize components as expected bu *XCTest* framework is not able to get these details.
They are not retrieved thus we cannot make automated tests on them.

**The aim is to see how we can get these details and make tests.**

For example, given the `OUDSButton`, its body is defined like:

```swift
    public var body: some View {
        // A button with negative or brand appearance is not allowed on a colored surface.
        // Test is done here because onColoredSurface is environment variable which is not accessible in init.
        if onColoredSurface, appearance == .negative || appearance == .brand {
            OL.fatal("An OUDSButton with OUDSButton.Appearance.{Negative | Brand} appearance has been detected as a direct or indirect child of an OUDSColoredSurface, which is not allowed.")
        }

        Button(action: action) {
            switch type {
            case let .icon(icon, _):
                ButtonIcon(icon: icon)
            case let .text(text):
                ButtonText(text: text)
            case let .textAndIcon(text, icon):
                ButtonTextAndIcon(text: text, icon: icon)
            }
        }
        .buttonStyle(OUDSButtonStyle(isHover: isHover, appearance: appearance, style: style))
        .disabled(style == .loading)
        .accessibilityLabel(accessibilityLabel)
        .onHover { isHover in
            self.isHover = isHover
        }
    }
```

In _Design System Toolbox_ we have the `ButtonDemo` like:
```swift
private struct ButtonDemo: View {

    @StateObject var configurationModel: ButtonConfigurationModel

    @Environment(\.theme) private var theme

    var body: some View {
        HStack(alignment: .center) {
            Spacer()

            // It is not allowed to place a Negative or Brand button on colored surface
            if configurationModel.onColoredSurface, configurationModel.appearance == .negative || configurationModel.appearance == .brand {
                Text("app_components_button_negative_hierary_notAllowed_text")
            } else {
                switch configurationModel.layout {
                case .iconOnly:
                    OUDSButton(icon: Image(decorative: "ic_heart"),
                               accessibilityLabel: "app_components_common_icon_a11y".localized(),
                               appearance: configurationModel.appearance,
                               style: configurationModel.style) {}
                        .accessibilityIdentifier(A11YIdentifiers.componentButton)
                case .textOnly:
                    OUDSButton(text: configurationModel.text,
                               appearance: configurationModel.appearance,
                               style: configurationModel.style) {}
                        .accessibilityIdentifier(A11YIdentifiers.componentButton)
                case .textAndIcon:
                    OUDSButton(icon: Image(decorative: "ic_heart"),
                               text: configurationModel.text,
                               appearance: configurationModel.appearance,
                               style: configurationModel.style) {}
                        .accessibilityIdentifier(A11YIdentifiers.componentButton)
                }
            }

            Spacer()
        }
        .disabled(!configurationModel.enabled)
        .padding(.all, theme.spaces.spaceFixedMedium)
    }
}
```

In the _XCTestCase_ we get the element with this method:
```swift
    @MainActor func buttons(withA11yIdentifier id: String, _ app: XCUIApplication) -> XCUIElementQuery {
        let elements = app.buttons.matching(identifier: id)
        XCTAssertTrue(elements.count > 0, "No matching button with accessibility identifier '\(id)'")
        return elements
    }
```

And here relies the test case method:
```swift
    @MainActor func testButtonA11yConfigurations() throws {
            
        let app = launchApp()
        goToComponentsSheet(app)
        tapButton(withWording: "app_components_button_label", app)
        let button = buttons(withA11yIdentifier: A11YIdentifiers.componentButton, app).firstMatch
        
        wait(2)
                
        print("@@@ a11y hint = \(button.accessibilityHint ?? "nil")")
        print("@@@ a11y label =  \(button.accessibilityLabel ?? "nil")")
        print("@@@ a11y value = \(button.accessibilityValue ?? "nil")")
        print("@@@ a11y traits = \(button.accessibilityTraits)\"")
        
        print("@@@ label = \(button.label)")
        print("@@@ label = \(button.value)")
        print("@@@ label = \(button.elementType)")

        debugAccessibilityHierarchy(button)
    }

    @MainActor func debugAccessibilityHierarchy(_ element: XCUIElement, level: Int = 0) {
        let indent = String(repeating: "  ", count: level)
        print("@@@ \(indent)Element: \(element.elementType) - Label: '\(element.label)' - Value: '\(element.value)'")
        print("@@@ \(indent)Element: \(element.accessibilityHint ?? "nil") - Label: '\(element.accessibilityValue ?? "nil")' - Value: '\(element.accessibilityTraits)'")
        
        for child in element.children(matching: .any).allElementsBoundByIndex {
            debugAccessibilityHierarchy(child, level: level + 1)
        }
    }
```

But, here are the results when we attempt to get the accessibility details:
```text
@@@ a11y hint = nil
@@@ a11y label =  nil
@@@ a11y value = nil
@@@ a11y traits = UIAccessibilityTraits(rawValue: 0)"
@@@ label = Button
@@@ label = Optional()
@@@ label = XCUIElementType(rawValue: 9)
@@@ Element: XCUIElementType(rawValue: 9) - Label: 'Button' - Value: 'Optional()'
@@@ Element: nil - Label: 'nil' - Value: 'UIAccessibilityTraits(rawValue: 0)'
```

i.e. not accessibility label, no accessibility value, no accessibility hint, in the element or in its only child.

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **0.21.0**
- Build number: **1762207613**
- Build tag: **0bd5fb8**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **0.21.0**
- Build number: **1762294009**
- Build tag: **25f875c**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pylapp** [comment]
📣 **New release available** 🚀

Please, get the [new release from GitHub](https://github.com/Orange-OpenSource/ouds-ios-design-system-toolbox/releases/tag/0.21.0) and wait for App Store update to test it!

- Display name: **Design System Toolbox**
- Version: **0.21.0**
- Build number: **1762362134**



_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_
